python/common/org/python/types/Bytes.java

```diff
@@ -971,10 +972,70 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
     }
 
     @org.python.Method(
-            __doc__ = "B.expandtabs(tabsize=8) -> copy of B\n\nReturn a copy of B where all tab characters are expanded using spaces.\nIf tabsize is not given, a tab size of 8 characters is assumed."
+            __doc__ = "B.expandtabs(tabsize=8) -> copy of B\n\nReturn a copy of B where all tab characters are expanded using spaces.\nIf tabsize is not given, a tab size of 8 characters is assumed.",
+            default_args = {"tabsize"}
     )
-    public org.python.Object expandtabs(java.util.List<org.python.Object> args, java.util.Map<java.lang.String, org.python.Object> kwargs, java.util.List<org.python.Object> default_args, java.util.Map<java.lang.String, org.python.Object> default_kwargs) {
-        throw new org.python.exceptions.NotImplementedError("bytes.expandtabs has not been implemented.");
+    public org.python.Object expandtabs(org.python.types.Object tabsize) {
+        // CPython implementation https://github.com/python/cpython/blob/master/Objects/stringlib/transmogrify.h#L21
+        // TODO: check overflows
+
+        if (this.value == null) {
+            return null;
+        }
+
+        if (this.value.length == 0) {
+            return new org.python.types.Bytes(new byte[0]);
+        }
+
+        int tabsize_int = 8;
+
+        if (tabsize != null) {
+            tabsize_int = (int) ((org.python.types.Int) tabsize).value;
+        }
+
+
+        int output_size = 0;
+        for (int i = 0, j = 0; i < this.value.length; i++) {
```

Comment:
I don't understand what this for loop is accomplishing, could you clarify?

It _looks_ like it's meant to compute the output_size, but the code below doesn't seem to be using it in a meaningful way.

Reply:
ah yes, I forgot to change the variable name, I didn't fully understand it in the beginning, then I went through the documentation. It adjusts the tab sizes using columns, details [here](https://docs.python.org/3/library/stdtypes.html#str.expandtabs)

Reply:
ah, I'm sorry about that I've forgotten it from an older version of the code, the C implementation first calculates the size of the output then makes a buffer accordingly, we don't need that now.